pilot/cmd/pilot-agent/main.go

```diff
@@ -279,9 +277,14 @@ var (
 			secOpts.InitialBackoffInMilliSec = int64(initialBackoffInMilliSecEnv)
 			// Disable the secret eviction for istio agent.
 			secOpts.EvictionDuration = 0
-			if citadel.ProvCert != "" {
-				secOpts.AlwaysValidTokenFlag = true
+			secOpts.AlwaysValidTokenFlag = (platform == security.K8S)
```

Comment:
That seems a breaking change - and pretty sure it's not correct. In K8S the token is not always valid AFAIK.